src/ast/semantic_analyser.cpp

```diff
@@ -1725,6 +1756,31 @@ void SemanticAnalyser::visit(AttachPoint &ap)
       }
     }
   }
+  else if (ap.provider == "kfunc" || ap.provider == "kretfunc")
+  {
+#ifdef HAVE_KFUNC
+    bool supported = bpftrace_.btf_.has_data();
+#else
+    bool supported = false;
+#endif
+
+    if (!supported)
+      error("kfunc/kretfunc not available for your kernel version.", ap.loc);
```

Comment:
```suggestion
{
      error("kfunc/kretfunc not available for your kernel version.", ap.loc);
      return;
}
```

To avoid a 'Failed to resolv...' error too

Reply:
ok